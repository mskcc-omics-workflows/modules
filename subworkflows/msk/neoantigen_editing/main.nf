include { NEOANTIGENEDITING_ALIGNTOIEDB      } from '../../../modules/msk/neoantigenediting/aligntoiedb'
include { NEOANTIGENEDITING_COMPUTEFITNESS   } from '../../../modules/msk/neoantigenediting/computefitness'


workflow NEOANTIGEN_EDITING {

    take:
    neoantigenInput_ch
    iedbfasta

    main:

    ch_versions = Channel.empty()

    NEOANTIGENEDITING_ALIGNTOIEDB (neoantigenInput_ch, iedbfasta)
    ch_versions = ch_versions.mix(NEOANTIGENEDITING_ALIGNTOIEDB.out.versions.first())


    ch_computeFitnessIn = neoantigenInput_ch.combine(NEOANTIGENEDITING_ALIGNTOIEDB.out.iedb_alignment, by: [0])

    NEOANTIGENEDITING_COMPUTEFITNESS ( ch_computeFitnessIn )

    ch_versions = ch_versions.mix(NEOANTIGENEDITING_COMPUTEFITNESS.out.versions.first())

    emit:
    annotated_output      = NEOANTIGENEDITING_COMPUTEFITNESS.out.annotated_output           // channel: [ val(meta), [ annotated_json ] ]
    versions              = ch_versions                                                     // channel: [ versions.yml ]
}
