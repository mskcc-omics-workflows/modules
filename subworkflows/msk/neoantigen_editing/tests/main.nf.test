nextflow_workflow {

    name "Test Workflow NEOANTIGEN_EDITING"
    script "../main.nf"
    workflow "NEOANTIGEN_EDITING"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/neoantigen_editing"
    tag "neoantigeninput"
    tag "neoantigenediting/aligntoiedb"
    tag "neoantigenediting/computefitness"


    test("neoantigen_editing - json") {

        when {
            workflow {
                """
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true)
                    ])
                input[1] = Channel.value(file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta'], checkIfExists: true))

                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.versions,
                    file(workflow.out.annotated_output[0][1]).name
                    ).match()
                }
            )

        }
    }

    test("neoantigen_editing - json - stub") {

        options "-stub"

        when {
            workflow {
                """
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map,
                    file('patient_data')
                    ])
                input[1] = Channel.value(file('iedb_fasta'))

                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.versions,
                    file(workflow.out.annotated_output[0][1]).name
                    ).match()
                }

            )
        }
    }
}
