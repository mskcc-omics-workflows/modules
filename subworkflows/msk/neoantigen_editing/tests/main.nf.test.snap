{
    "neoantigen_editing - json - stub": {
        "content": [
            [
                "versions.yml:md5,03ac4fd1eeae826979dfbed8a7b3186c",
                "versions.yml:md5,3ce37ca54350a7f641e05f0de9f87f7f"
            ],
            "patient_data_annotated.json"
        ],
        "timestamp": "2024-06-19T14:59:58.874516"
    },
    "neoantigen_editing - json": {
        "content": [
            [
                "versions.yml:md5,03ac4fd1eeae826979dfbed8a7b3186c",
                "versions.yml:md5,3ce37ca54350a7f641e05f0de9f87f7f"
            ],
            "3_OLTS_primary_tumor_annotated.json"
        ],
        "timestamp": "2024-06-19T14:59:52.198482"
    }
}